packages/notte-browser/src/notte_browser/session.py

```diff
@@ -436,7 +439,13 @@ async def scrape(
         **scrape_params: Unpack[ScrapeParamsDict],
     ) -> DataSpace:
         if url is not None:
-            _ = await self.goto(url)
+            try:
+                _ = await self.goto(url)
+            except PageLoadingError:
```

Comment:
The `self.goto(url) fails in 100% of cases when the url ends with `.pdf` right ? If that's correct, then you probably want to move the if statement before the goto and only run the goto in url does not end with .pdf